orttraining/orttraining/training_api/checkpoint.cc

```diff
@@ -550,10 +551,18 @@ Status OrtLoadInternal(const PathString& checkpoint_path,
     const auto tensor_file_full_path = ConcatPathComponent<PathChar>(checkpoint_path, tensor_file_path);
     LoadTensorProtoFromFile(tensor_file_full_path, tensor_protos, "[params]");
 
-    for (const auto& tensor_proto : tensor_protos) {
-      param_tensor_protos.push_back(tensor_proto);
+    for (auto& tensor_proto : tensor_protos) {
+      param_tensor_protos.insert(std::make_pair(tensor_proto.name(), tensor_proto));
     }
   }
+
+  // Load imported initializers into the Model
+  for (auto& init : *(model_proto.mutable_graph()->mutable_initializer())) {
+    if (!init.has_name()) ORT_THROW("Initializer name is missing.");
+    auto it = param_tensor_protos.find(init.name());
+    init = it->second;
```

Comment:
Add a check to validate that the matching initializer was found in param_tensor_protos.